.github/workflows/create-hotfix-branch.yml

```diff
@@ -12,10 +14,44 @@ jobs:
     name: Create New Branch
     runs-on: ubuntu-latest
 
-    # Only allow release stakeholders to initiate releases
-    if: github.ref == 'refs/heads/main' && (github.actor == 'ItsSudip' || github.actor == 'krishna2020' || github.actor == 'lvrach' || github.actor == 'cisse21' || github.actor == 'debanjan97' || github.actor == 'saikumarrs' || github.actor == 'sandeepdsvs' || github.actor == 'utsabc' || github.actor == 'yashasvibajpai' || github.actor == 'maheshkutty' || github.actor == 'manish339k') && (github.triggering_actor == 'ItsSudip' || github.triggering_actor == 'krishna2020' || github.triggering_actor == 'lvrach' || github.triggering_actor == 'cisse21' || github.triggering_actor == 'debanjan97' || github.triggering_actor == 'saikumarrs' || github.triggering_actor == 'sandeepdsvs' || github.triggering_actor == 'utsabc' || github.triggering_actor == 'yashasvibajpai' || github.triggering_actor == 'maheshkutty' || github.triggering_actor == 'manish339k')
     steps:
+      - name: Check team membership or allowlist
+        id: check_access
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const allowedActors = [
+              "ItsSudip", "krishna2020", "lvrach", "cisse21", "debanjan97",
+              "saikumarrs", "sandeepdsvs", "utsabc", "yashasvibajpai",
+              "maheshkutty", "manish339k"
+            ];
+            const org = context.repo.owner;
+            const team_slug = "@rudderlabs/integrations-platform"; // <-- update to your team slug
+            const actor = context.actor;
```

Comment:
Have you checked that this gives the actor information even when the workflow is manually re-run?
`${{ github.actor || github.triggering_actor }}` - typically this covers it.